Frameworks/CoreGraphics/CGDataProvider.mm

```diff
@@ -156,16 +421,28 @@ void CGDataProviderRelease(CGDataProviderRef data) {
  @Notes
 */
 CGDataProviderRef CGDataProviderRetain(CGDataProviderRef provider) {
-    CFRetain((id)provider);
-
+    CFRetain(provider);
     return provider;
 }
 
 /**
- @Status Stub
+ @Status Interoperable
  @Notes
 */
 CFTypeID CGDataProviderGetTypeID() {
-    UNIMPLEMENTED();
-    return StubReturn();
-}
\ No newline at end of file
+    return __CGDataProvider::GetTypeID();
+}
+
+#pragma endregion // CGDataProvider Functions
+
+#pragma region Internal CGDataProvider Functions
+
+const void* _CGDataProviderGetData(CGDataProviderRef provider) {
+    return provider ? provider->GetData() : nullptr;
+}
+
+size_t _CGDataProviderGetSize(CGDataProviderRef provider) {
+    return provider ? provider->GetSize() : 0;
```

Comment:
let's do the right thing :)

---
In reply to: [93116128](https://github.com/Microsoft/WinObjC/pull/1579#discussion_r93116128) [](ancestors = 93116128,93115787,93115443,92931590)